huaweicloud/services/css/resource_huaweicloud_css_logstash_pipeline.go

```diff
@@ -0,0 +1,411 @@
+	hotStartNames []string) error {
+	for _, name := range hotStartNames {
+		hotStartPipelineHttpUrl := "v1.0/{project_id}/clusters/{cluster_id}/lgsconf/hot-start"
+		hotStartPipelinePath := cssV1Client.Endpoint + hotStartPipelineHttpUrl
+		hotStartPipelinePath = strings.ReplaceAll(hotStartPipelinePath, "{project_id}", cssV1Client.ProjectID)
+		hotStartPipelinePath = strings.ReplaceAll(hotStartPipelinePath, "{cluster_id}", d.Get("cluster_id").(string))
+
+		hotStartPipelineOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+		}
+
+		hotStartPipelineOpt.JSONBody = map[string]interface{}{
+			"name":       name,
+			"keep_alive": d.Get("keep_alive").(bool),
+		}
+
+		_, err := cssV1Client.Request("POST", hotStartPipelinePath, &hotStartPipelineOpt)
+		if err != nil {
+			return fmt.Errorf("error hot start CSS logstash cluster pipeline: %s", err)
+		}
+
+		checkErr := startPipelineStatusCheck(ctx, d, cssV1Client)
+		if checkErr != nil {
+			return checkErr
+		}
+	}
+
+	return nil
+}
+
+func startPipelineStatusCheck(ctx context.Context, d *schema.ResourceData, cssV1Client *golangsdk.ServiceClient) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{startUpStatus},
+		Target:       []string{workingStatus},
+		Refresh:      startPipelineStateRefreshFunc(d, cssV1Client),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for the CSS logstash cluster pipeline to start: %s", err)
+	}
+	return nil
+}
+
+func stopPipelineStatusCheck(ctx context.Context, d *schema.ResourceData, cssV1Client *golangsdk.ServiceClient) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{stoppingStatus},
+		Target:       []string{stoppedStatus},
+		Refresh:      stopPipelineStateRefreshFunc(d, cssV1Client),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for the CSS logstash cluster pipeline to stop: %s", err)
+	}
+	return nil
+}
+
+func startPipelineStateRefreshFunc(d *schema.ResourceData, cssV1Client *golangsdk.ServiceClient) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		pipelines, err := getPipelines(d, cssV1Client)
+		if err != nil {
+			return pipelines, "ERROR", err
+		}
+		statusList := utils.ExpandToStringList(utils.PathSearch("[].status", pipelines, nil).([]interface{}))
+		if utils.IsStrContainsSliceElement(startUpStatus, statusList, false, false) {
+			return nil, startUpStatus, nil
+		}
+		return pipelines, workingStatus, nil
+	}
+}
+
+func stopPipelineStateRefreshFunc(d *schema.ResourceData, cssV1Client *golangsdk.ServiceClient) resource.StateRefreshFunc {
```

Comment:
this  func can be merged with the func of startPipelineStateRefreshFunc, workingStatus and stoppingStatus can be replaced with penddingStatus